lib/Twig/Node/Module.php

```diff
@@ -80,16 +80,7 @@ protected function compileTemplate(Twig_Compiler $compiler)
 
         $this->compileClassHeader($compiler);
 
-        if (
-            count($this->getNode('blocks'))
-            || count($this->getNode('traits'))
-            || !$this->hasNode('parent')
-            || $this->getNode('parent') instanceof Twig_Node_Expression_Constant
-            || count($this->getNode('constructor_start'))
-            || count($this->getNode('constructor_end'))
-        ) {
-            $this->compileConstructor($compiler);
-        }
+        $this->compileConstructor($compiler);
```

Comment:
Always calling `compileConstructor()` is actually another small performance optimization as it sets `$this->parent` to `false` when possibler, which avoids determining the parent during template evaluation.